luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/LuthierIndustrialPark.java

```diff
@@ -159,15 +168,56 @@ public DataSourceMetadataService getMetadataService() {
      *
      * @return the LogicalMetric instance corresponding to this metricName.
      */
-    // TODO: to be finished in the metric PR.
     public LogicalMetric getMetric(String metricName) {
+        // TODO: to be finished in the metric PR.
         return null;
     }
 
+    /**
+     * Retrieve or build a LogicalTable.
+     *
+     * @param tableId the identifier for table name and granularity to be retrieved or built
+     *
+     * @return the logicalTable instance corresponding to this TableIdentifier.
+     */
+    public LogicalTable getLogicalTable(TableIdentifier tableId) {
+        LogicalTableDictionary logicalTableDictionary = resourceDictionaries.getLogicalDictionary();
+        if (! logicalTableDictionary.containsKey(tableId)) {
+            String tableName = tableId.getKey() + tableId.getValue().toString();
```

Comment:
How is this even working? If we have a table with name `table1` and granularities `day`, `week`, `month`,
then we're creating the following groups of logical tables: `table1day, table1week, table1month` *each* with
granularities `day, week, month`, when we should be creating a group of logical tables : `table1` with
granularities `day, week, month`.

Reply:
This is working because it is not used to load the logicalTable. When the system populates the logicalTables, it is using the 
`private void putLogicalTableByGroup(String tableKey) `
a couple of lines below instead. While this getter is public, and is exposed to the users, the private function is used to internally read from json and populate the config source.